File: app/src/main/java/the/one/demo/ui/fragment/CategoryFragment.java
package the.one.demo.ui.fragment;

import android.view.View;
import android.widget.TextView;

import java.util.ArrayList;

import the.one.base.base.fragment.BaseFragment;
import the.one.base.base.fragment.BaseTitleTabFragment;
import the.one.demo.Constant;
import the.one.demo.R;


//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛

/**
 * @author The one
 * @date 2019/3/4 0004
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
public class CategoryFragment extends BaseTitleTabFragment {

    @Override
    protected void initView(View rootView) {
        super.initView(rootView);
//        把继承改为BaseTabOnTitleFragment注释掉下面的代码试试
        TextView mTitle =  mTopLayout.setTitle("GankType");
        mTitle.setTextColor(getColorr(R.color.qmui_config_color_gray_1));
        mTitle.getPaint().setFakeBoldText(true);
        // 这里不用注释掉,更换继承这里要改成true(是否有分割线)
        mTopLayout.setBackgroundDividerEnabled(false);
        mTopLayout.setBackgroundColor(getColorr(R.color.white));
    }

    @Override
    protected void addTabs() {
        for (int i = 0; i < Constant.title.length; i++) {
            if (Constant.title[i].equals(Constant.IOS))
                addTab("IOS");
            else
                addTab(Constant.title[i]);
        }
    }

    @Override
    protected void addFragment(ArrayList<BaseFragment> fragments) {
        for (int i = 0; i < Constant.title.length; i++) {
            fragments.add(GankFragment.newInstance(Constant.title[i]));
        }
    }
}

File: kotlin/src/main/java/the/one/kotlin/fragment/CategoryFragment.kt
package the.one.kotlin.fragment

import android.view.View
import the.one.base.base.fragment.BaseFragment
import the.one.base.base.fragment.BaseTitleTabFragment
import the.one.kotlin.Constant
import java.util.*


//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛
/**
 * @author The one
 * @date 2019/4/1 0001
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */

class CategoryFragment: BaseTitleTabFragment(){

    override fun initView(rootView: View?) {
        super.initView(rootView)
        mTopLayout.setTitle("GankType").paint.isFakeBoldText = true
        mTopLayout.setBackgroundDividerEnabled(false)
    }

    override fun addTabs() {
        for (i in Constant.title.indices) {
            if (Constant.title[i] == Constant.IOS)
                addTab("IOS")
            else
                addTab(Constant.title[i])
        }
    }

    override fun addFragment(fragments: ArrayList<BaseFragment>?) {
        for (i in Constant.title.indices) {
            fragments?.add(GankFragment.newInstance(Constant.title[i]))
        }
    }

}